### ISSUE TITLE
Wrong startup/settings/loading values

### ISSUE THREAD

**Issue by lukaville** [nebula-plugins/gradle-metrics-plugin#50]
Hi! I'm applying the plugin this way:

```groovy
buildscript {
    ext.kotlin_version = '1.3.11'
    repositories {
        google()
        jcenter()
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.3.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath "com.netflix.nebula:gradle-metrics-plugin:8.3.3"
    }
}

apply plugin: "nebula.metrics"
metrics {
    dispatcherType = 'REST'
    restUri = 'https://webhook.site/6bd45598-0a99-44fd-ba9d-6e1ddf1c758d'      // default is 'http://localhost/metrics'
    restBuildEventName = 'my_build_events'            // default is 'build_metrics'
    restLogEventName = 'my_log_events'                // default is 'build_logs'

    headers['myHeader_key'] = 'header_value'          // optional, default is empty MAP object
}
```

And it reports incorrect startup/settings/loading values (they are negative or 0):
```json
"events": [
    {
      "description": "startup",
      "type": "init",
      "elapsedTime": -7270
    },
    {
      "description": "settings",
      "type": "configure",
      "elapsedTime": -1548243439147
    },
    {
      "description": "projectsLoading",
      "type": "configure",
      "elapsedTime": 0
    },
    {
      "description": ":",
      "type": "configure",
      "elapsedTime": 1548243446424
    },
    {
      "description": ":app",
      "type": "configure",
      "elapsedTime": 911
    },
    {
      "description": ":app:classpath",
      "type": "resolve",
      "elapsedTime": 0
    },
    {
      "description": "task",
      "type": "execution",
      "elapsedTime": 414
    },
    {
      "description": "task",
      "type": "execution",
      "elapsedTime": 0
    }
  ],
```

Looks like it's happening because projectsLoaded callback in GradleBuildMetricsCollector (and all previous callbacks) is called before applying the plugin. Is it intended or there are ways to apply the plugin differently to make it work?

Thank you

**Comment by sophiataskova** [comment]
I'm seeing the same thing. I wonder if this plugin just isn't supported anymore :(

**Comment by chali** [comment]
Hello folks, no the plugin is still supported. I just checked and our builds are having the same issue.

**Comment by rpalcolea** [comment]
Hi folks,

We found that `BuildAdapter` and `ProjectEvaluationListener` methods related to project/settings evaluation are not triggered anymore when applying the plugin in Gradle 5.+

This is because this listeners shouldn't be applied at `Project` level but `Gradle` or `Settings` level. 

We just released 10.0.0 of the plugin which introduces `nebula.metrics.settings-plugin` and `nebula.metrics.init-plugin`

In order to measure `settingsEvaluated`, `projectsLoaded` and others, the plugin should be applied now in settings.gradle or init.gradle.

We, use the `init.gradle` approach as we do custom internal distribution of Gradle (Nebula). This is what we do now:

`init.gradle`
```
initscript {
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
  }
  dependencies {
    classpath "com.netflix.nebula:gradle-metrics-plugin:10.0.0"
  }
}

apply plugin:  nebula.plugin.metrics.MetricsInitPlugin
```

We haven't used with `settings.gradle`, but similar to `build-scans` plugin. you should be able to do the following:

```
plugins {
  id "nebula.metrics.settings-plugin" version "10.0.0"
}
```
